go/interpreter/lfvm/instruction_statistics.go

```diff
@@ -24,20 +24,26 @@ type statisticRunner struct {
 	stats *statistics
 }
 
-func (s *statisticRunner) run(c *context) {
+func (s *statisticRunner) run(c *context) (status, error) {
 	stats := statsCollector{stats: newStatistics()}
-	for c.status == statusRunning {
+	status := statusRunning
+	var err error
+	for status == statusRunning {
 		if c.pc < int32(len(c.code)) {
 			stats.nextOp(c.code[c.pc].opcode)
 		}
-		step(c)
+		status, err = step(c)
+		if err != nil {
+			return status, err
```

Comment:
given this is the `statisticRunner` I would do a `break` here so that stats can be inserted, and at the end return `status, err`, since it should be nil if all is fine. no? wdyt?

Reply:
yes, you are right.